src/main/java/org/openlmis/buq/service/buq/BottomUpQuantificationService.java

```diff
@@ -721,6 +726,38 @@ private List<ProductGroupsCostData> createProductsCostData(boolean isDistrictLev
     return productsCostsList;
   }
 
+  private boolean checkFacilityTypeAndPermission(MinimalFacilityDto facility) {
+    String checkedFacilityTypeName = "Health Centre";
+    UserDto user = authenticationHelper.getCurrentUser();
+    RightDto mohRight = rightReferenceDataService.findRight(MOH_APPROVAL_RIGHT_NAME);
+    ResultDto<Boolean> hasMohRight = new ResultDto<>();
+    if (mohRight != null) {
+      hasMohRight = userReferenceDataService
+          .hasRight(user.getId(), mohRight.getId(), null, null, null);
+    } else {
+      hasMohRight.setResult(false);
+    }
+
+    RightDto poralgRight = rightReferenceDataService.findRight(PORALG_APPROVAL_RIGHT_NAME);
+    ResultDto<Boolean> hasPoralgRight = new ResultDto<>();
+    if (poralgRight != null) {
+      hasPoralgRight = userReferenceDataService
+          .hasRight(user.getId(), poralgRight.getId(), null, null, null);
+    } else {
+      hasPoralgRight.setResult(false);
+    }
+
+    if (hasMohRight.getResult() && hasPoralgRight.getResult()) {
+      return true;
+    } else if (Boolean.TRUE.equals(hasPoralgRight.getResult())) {
+      return facility.getType().getName().equals(checkedFacilityTypeName);
+    } else if (Boolean.TRUE.equals(hasMohRight.getResult())) {
+      return !facility.getType().getName().equals(checkedFacilityTypeName);
+    } else {
+      return false;
+    }
+  }
```

Comment:
nitpicky, no change request:
`else` is not required after `return`, so it'd be enough to do:
```java
if (hasMohRight.getResult() && hasPoralgRight.getResult()) {
      return true;
}
if (Boolean.TRUE.equals(hasPoralgRight.getResult())) {
      return facility.getType().getName().equals(checkedFacilityTypeName);
}
if (Boolean.TRUE.equals(hasMohRight.getResult())) {
      return !facility.getType().getName().equals(checkedFacilityTypeName);
}
return false;
```

Reply:
else statement removed